staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -32,19 +37,26 @@ type attributesRecord struct {
 	object      runtime.Object
 	oldObject   runtime.Object
 	userInfo    user.Info
+
+	annotations map[string]string
+	// other elements are always accessed in single goroutine.
+	// But ValidatingAdmissionWebhook sets annotations concurrently.
```

Comment:
seems like ValidatingAdmissionWebhook should do its own locking if it is the only actor operating concurrently on this object

Reply:
I suppose all validating admission could be called in parallel in the future, not just ValidatingAdmissionWebhook. making SetAnnotation threadsafe is probably fine.